fairscale/utils/testing.py

```diff
@@ -230,10 +243,10 @@ def replacement(*args: Any, **kwargs: Any) -> None:
             if "OMPI_COMM_WORLD_RANK" in os.environ:
                 os.environ["RANK"] = os.environ["OMPI_COMM_WORLD_RANK"]
                 os.environ["WORLD_SIZE"] = os.environ["OMPI_COMM_WORLD_SIZE"]
-                os.environ["MASTER_ADDR"] = "localhost"
-                os.environ["MASTER_PORT"] = "10638"
-                torch.distributed.init_process_group("mpi")
+                _, filename = tempfile.mkstemp()
```

Comment:
@anj-s could that be the faulty part ? looks wrong to me actually now, each rank will get a different file to sync on

Reply:
Looks like this might be the issue. The file that workers sync on needs to be on a shared file system [doc](http://man.hubwiz.com/docset/PyTorch.docset/Contents/Resources/Documents/distributed.html#shared-file-system-initialization) Given the timeout issue which points to workers not connecting to each other, this would make sense.

CircleCI provides a way to share metadata which we can perhaps use for this. I'll try changing this for the current test.

Looking at the previous version of this code it looks like ENV vars were used. It would still require the rank 0 worker IP and port for all processes to connect across workers. If this is on a single machine which I think is the case, can we move to using ENV vars again? Any reason why we prefer a file based approach?

Reply:
Another thing that the file based change does not answer is the flakiness. I would have expected this to always fail and for all torch versions.

Reply:
the issue with env variables is that all instances of these tests use the same port, so they often collide with another if the test infra is concurrent (which FB infra is). I would really prefer to stick to a file, it's just that it needs to be shared across the ranks (like for c10 and rpc init), instead of being done here.

Reply:
> Another thing that the file based change does not answer is the flakiness. I would have expected this to always fail and for all torch versions.

yep, this I don't know

Reply:
proposed fix in https://github.com/facebookresearch/fairscale/pull/316